src/app/qgisapp.cpp

```diff
@@ -5643,16 +5643,20 @@ void QgisApp::fileOpenAfterLaunch()
     return;
   }
 
-  if ( !projPath.endsWith( QLatin1String( ".qgs" ), Qt::CaseInsensitive ) &&
-       !projPath.endsWith( QLatin1String( ".qgz" ), Qt::CaseInsensitive ) )
+  // Is this a storage based project?
+  const bool projectIsFromStorage { QgsApplication::instance()->projectStorageRegistry()->projectStorageFromUri( projPath )  };
+
+  if ( !( projectIsFromStorage ||
+          projPath.endsWith( QLatin1String( ".qgs" ), Qt::CaseInsensitive ) ||
+          projPath.endsWith( QLatin1String( ".qgz" ), Qt::CaseInsensitive ) ) )
```

Comment:
A particular reason to change from `!a && !b && !c` to `!( a || b || c )`? Maybe it's just me, but I find the first easier to read.

Reply:
Matter of tastes I guess.

Reply:
Not sure on that... Sequential parsing is probably easier than than inverting a grouped operation. Or something along these lines.
Anyway, not gonna stop this nice pull request on this one.